src/library/searchqueryparser.h

```diff
@@ -11,14 +11,19 @@
 
 class SearchQueryParser {
   public:
-    SearchQueryParser(QSqlDatabase& database);
+    SearchQueryParser(TrackCollection* pTrackCollection);
+    SearchQueryParser(QSqlDatabase& database)
```

Comment:
This constructor shouldn't be available anymore! Otherwise m_pTrackCollection is uninitialized.